### DISCUSSION TITLE
Conditional Config Contexts

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#16195] [Q&A]
Hey there fellows!

I'm working on some config contexts.
I'm using it with association to clusters.

Is there a way to make a config context conditional to a cluster?

like my config context lets say is:

```
{
           "connections"" [
                {
                       "interfaces: [
                              "interface A" ]                               
                }
                                     ]
}
```

I want it to be something like:
```
if cluster a:
{
           "connections"" [
                {
                       "interfaces: [
                              "interface A" ]                               
                }
                                     ]
}
if cluster b:

{
           "connections"" [
                {
                       "interfaces: [
                              "interface b" ]                               
                }
                                     ]
}
```

I suppose this is not possible? But is there something close? some way around it? perhaps 2 `Data`s in the config context?

It is important because we have 2 types of servers in the same clusters one with 4 interfaces and 1 with only 2 but they are the same cluster.

**Comment by candlerb** [comment]
> Is there a way to make a config context conditional to a cluster?

Yes.  Create two separate config contexts.  Assign one of them to cluster A, and one to cluster B; this is in the "Assignments" section at the bottom of the config context creation page (`/extras/config-contexts/add/`)

![image](https://github.com/netbox-community/netbox/assets/44789/69a37dd4-9e60-4595-b104-51913af0c5a8)

When you derive the overall config context for a given VM, it will get either the cluster A or cluster B data merged into its context, depending on which cluster it belongs to.

> It is important because we have 2 types of servers in the same clusters one with 4 interfaces and 1 with only 2 but they are the same cluster.

Then you are asking for something completely different: you want the configuration data to be conditional on the device, not the cluster.

Although it's possible to assign a VM to a specific device in a cluster, that assignment cannot be used for selecting a config context (as far as I can see).

If your cluster is not homogenous, then I think you should model it as two separate clusters: one which has one set of interfaces, and one another. After all, you probably cannot migrate VMs between these nodes anyway.

**Reply by remchuk** [reply]
You are correct I have mixed things up a bit, sorry.

What I have is one cluster, cluster Alpha.

Cluster Alpha is only devices, no VMs.
There are 2 types of Servers in the cluster.

The First one is with 2 interfaces .
The second one is 4 interfaces.

It is quite important that both of those types will be under the same cluster.
Because they are both cluster Alpha.
